### ISSUE TITLE
Intelligent interaction with dist-tags - automatic tagging

### ISSUE THREAD

**Issue by egucciar** [semantic-release/semantic-release#866]
## New feature motivation
It would be nice if semantic-release could work on two branches together and coordinate dist-tags. This allows us to create releases on channels besides latest, but have those same releases automatically updated to latest once the branch is merged to another branch.

## New feature description
Here's an example - there is a **staging/develop/beta** branch and a **master/stable** branch. Whenever anything is merged into _staging_, a new release is provisioned by looking at the git-tags and doing the normal publishing but with a **dist-tag besides latest - i.e. beta**. Then, when this branch is merged to **master** it will find the last release (via git tag) from the incoming branch, and then **update the dist-tag for that release to latest**.

**Comment by pvdlg** [comment]
I think it's a duplicate of #563?

**Comment by egucciar** [comment]
@pvdlg I am interested in trying this out because it's currently a blocker in our Dev workflow to not have this sort of capability. We currently have this setup:

`master` branch is pre-existing and running semantic-release in circleCI and contains the latest gitTag of 4.x

`staging` branch is pre-existing and was branched from `master` at some point in time. the last git tag on the branch is 3.x

We want to use `staging` to create a *pre-release* which can be installed explicitly by consuming apps with the `--beta` tag. Then, when staging is merged into master, we want there to be an actual release. Ideally, the same `--beta` release could have its `dist-tag` updated to `--latest` but it could also be an issue to do this considering the source code could theoretically be different after a merge.

I tried your branch - Thanks so much for this by the way, incredible work! and I am running into the folowing:

```
[Semantic release]: The release type for the commit is major
[Semantic release]: Analysis of 473 commits complete: major release
[Semantic release]: The next release version is 3.0.0
[Semantic release]: EINVALIDNEXTVERSION TODO
TODO


{ SemanticReleaseError: TODO
    at module.exports (/Users/ericagucciardo/git/moveiq-app/node_modules/semantic-release/lib/get-error.js:6:10)
    at run (/Users/ericagucciardo/git/moveiq-app/node_modules/semantic-release/index.js:129:11)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
  name: 'SemanticReleaseError',
  code: 'EINVALIDNEXTVERSION',
  details: 'TODO',
  semanticRelease: true }
```

Any ideas on what I need to do next-steps wise to get this working for our needs? It would be lovely. Our use case currently is that our `master` branch is used to create Releases for our app for consumers, but it's also how we release our app to our site (since our App is consumed both by another consuming app, as well as deployed directly to a host from CI/CD). This means in order for us to QA the app in our consuming app, we have to create a production release, which can create delays when trying to move quickly since we need to completely QA things to push to Prod before we can consider QAing the staging on the consumer application. So, though there's multiple ways to solve this problem (i.e. having to merge to 2 different branches rather than having `master` serve the purpose of both), I think an even better solution would be to leverage Semantic Release to create `beta` releases for `staging` and `latest` releases for `master`. This would be the ideal workflow.
